packages/publisher/gcs/src/PublisherGCS.ts

```diff
@@ -23,24 +23,26 @@ export default class PublisherGCS extends PublisherStatic<PublisherGCSConfig> {
   async publish({ makeResults, setStatusLine }: PublisherOptions): Promise<void> {
     const artifacts: GCSArtifact[] = [];
 
-    if (!this.config.bucket) {
+    const { storageOptions, bucket: configBucket, folder, ...uploadOptions } = this.config;
```

Comment:
question: I haven't tested this locally yet but we do have a custom `keyResolver` property in the Config as well that would get rolled up with `...uploadOptions`. Would passing in that unknown key into `bucket.upload` trigger an error on the GCS SDK side?

Reply:
I haven't had any trouble with it, and it doesn't look like there's any checking for extraneous keys in [their library](https://github.com/googleapis/nodejs-storage/blob/c267e6b81e8d0d19f6c7ef886c6897e4bfc7897b/src/bucket.ts#L4353). That could change, of course.

We could also pull out `keyResolver` here if you think it's a concern